biz.aQute.bndlib/src/aQute/bnd/service/result/Result.java

```diff
@@ -225,4 +234,12 @@
 	 */
 	Result<V, E> recover(FunctionWithException<? super E, ? extends V> recover);
 
+	/**
+	 * Terminal function that processes the result or the error
+	 *
+	 * @param ok the consumer called when ok
+	 * @param err the consumer called when not ok
+	 */
+	void then(ConsumerWithException<? super V> ok, ConsumerWithException<? super E> err);
```

Comment:
I would call this something like `accept` rather than `then`. `then` implies some chaining in a monadic style. This is really about consuming the Result and `accept` is the Java method name for consumers.